src/ParamHandler.ts

```diff
@@ -104,9 +106,19 @@ export class ParamHandler {
     private parseValue(value: any, paramMetadata: ParamMetadata) {
         try {
             const parseValue = typeof value === "string" ? JSON.parse(value) : value;
-            if (paramMetadata.format !== Object && paramMetadata.format && this.driver.useClassTransformer) {
-                const options = paramMetadata.classTransformOptions || this.driver.plainToClassTransformOptions;
-                return plainToClass(paramMetadata.format, parseValue, options);
+            if (paramMetadata.format !== Object && paramMetadata.format) {
+                if (this.driver.useParamValidator) {
+                    const options = paramMetadata.paramValidatorOptions || this.driver.paramValidatorOptions;
+                    return transformAndValidate(paramMetadata.format, parseValue, options)
+                        .catch(err => {
+                            const error: any = new BadRequestError(`Invalid ${paramMetadata.type}, check 'errors' property for more info`);
+                            error.errors = err;
```

Comment:
59801f9 made this code more readable? or you have an other idea how the error handler should return correct http code and nice JSON with info?